### DISCUSSION TITLE
How to get the page's current hash live.

### DISCUSSION THREAD

**Discussion by JamDev0** [vercel/next.js#58565] [Help]
### Summary

I was developing a navbar, and needed to check whether its links were active or not, counting that, besides slash-links `href='/login'` they could also be hash-links `href='#aboutUs'`. 

After a lot of time, I came to the following solution (navbar's links component code):

``` TypeScript
'use client'

import Link from "next/link";
import { useSearchParams } from "next/navigation";
import { useEffect, useMemo, useState } from "react";

interface NavLinkProps {
  href: string;
  title: string;
}

export function NavLink({ href, title }: NavLinkProps) {
	// Used to trigger the side effect responsible of updating the path state  
	const refreshCurrentPath = useSearchParams();

  const isAnchorLink = href.at(0) === '#';

  const [path, setPath] = useState(isAnchorLink 
	  ?
			decodeURIComponent(window.location.hash) 
		: 
			window.location.pathname
	);

  const isActive = useMemo(() => path === href, [path, href])

  useEffect(() => {
    setPath(isAnchorLink 
	    ? 
		    decodeURIComponent(window.location.hash) 
			: 
				window.location.pathname
		)
  }, [refreshCurrentPath, isAnchorLink])
  
  return (

    <Link 
	    key={title} 
	    href={href} 
	    data-active={isActive} 
	    className="text-red-400 data-[active=true]:text-blue-400"
    >
      {title}
    </Link>
  )
}
```

Needing to use the `useSearchParams` to do an update the component, seems a little hacky, so there must be a better way of doing so, you guys have any idea?

**Comment by icyJoseph** [comment]
You can use the `hashchanged` event I guess, it depends on your intention. Long ago I also did this with hooks, but your "live" requirement, makes me thing more of synchronizing with the native event:

- https://developer.mozilla.org/en-US/docs/Web/API/Window/hashchange_event
